wnfs/src/private/file.rs

```diff
@@ -62,29 +108,370 @@ impl PrivateFile {
+                        let bytes = Self::decrypt_block(key, &label, hamt, store).await?;
+                        yield bytes
+                    }
+                }
+            }
+        })
+    }
+
+    /// Gets the entire content of a file.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use std::rc::Rc;
+    /// use chrono::Utc;
+    /// use rand::thread_rng;
+    /// use wnfs::{
+    ///     private::{PrivateForest, PrivateRef},
+    ///     MemoryBlockStore, Namefilter, PrivateFile,
+    ///     utils::get_random_bytes, MAX_BLOCK_SIZE
+    /// };
+    ///
+    /// #[async_std::main]
+    /// async fn main() {
+    ///     let store = &mut MemoryBlockStore::default();
+    ///     let rng = &mut thread_rng();
+    ///     let hamt = Rc::new(PrivateForest::new());
+    ///
+    ///     let content = get_random_bytes::<100>(rng).to_vec();
+    ///     let (file, hamt) = PrivateFile::with_content(
+    ///         Namefilter::default(),
+    ///         Utc::now(),
+    ///         content.clone(),
+    ///         hamt,
+    ///         store,
+    ///         rng,
+    ///     )
+    ///     .await
+    ///     .unwrap();
+    ///
+    ///     let mut all_content = file.get_content(&hamt, store).await.unwrap();
+    ///
+    ///     assert_eq!(content, all_content);
+    /// }
+    /// ```
+    pub async fn get_content<B: BlockStore>(
+        &self,
+        hamt: &PrivateForest,
+        store: &B,
+    ) -> Result<Vec<u8>> {
+        let mut content = Vec::with_capacity(self.get_content_size_upper_bound());
+        let mut stream = self.stream_content(hamt, store);
+        while let Some(bytes) = stream.next().await {
+            content.extend_from_slice(&bytes?);
+        }
+        Ok(content)
+    }
+
+    /// Gets the block at specified index.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use std::rc::Rc;
+    /// use chrono::Utc;
+    /// use rand::thread_rng;
+    /// use wnfs::{
+    ///     private::{PrivateForest, PrivateRef},
+    ///     MemoryBlockStore, Namefilter, PrivateFile,
+    ///     utils::get_random_bytes, MAX_BLOCK_SIZE
+    /// };
+    ///
+    /// #[async_std::main]
+    /// async fn main() {
+    ///     let store = &mut MemoryBlockStore::default();
+    ///     let rng = &mut thread_rng();
+    ///     let hamt = Rc::new(PrivateForest::new());
+    ///
+    ///     let content = get_random_bytes::<100>(rng).to_vec();
+    ///     let (file, hamt) = PrivateFile::with_content(
+    ///         Namefilter::default(),
+    ///         Utc::now(),
+    ///         content.clone(),
+    ///         hamt,
+    ///         store,
+    ///         rng,
+    ///     )
+    ///     .await
+    ///     .unwrap();
+    ///
+    ///     let mut content_at_0 = file.get_block_at(0, &hamt, store).await.unwrap();
+    ///
+    ///     assert_eq!(content, content_at_0.unwrap());
+    /// }
+    /// ```
+    pub async fn get_block_at<B: BlockStore>(
```

Comment:
What do you think of a `seek` API here?
I.e. `seek(&self, offset: usize, limit: usize) -> Option<Vec<u8>>`, roughly?
It would return `None` if `offset >= file_size` and have the invariant that `vec.len() <= limit`, if the result is `Some(vec)`.

I guess we could have both, and `seek` could be implemented in terms of `get_block_at`?
Or we could also have `stream_from` take a block index as offset, so we would stream blocks starting at that block index, which could be the ultimate underlying function?

Thoughts?

Reply:
What would be the benefit of a limit in bytes?